pandas/core/arrays/categorical.py

```diff
@@ -1740,6 +1740,12 @@ def take_nd(self, indexer, allow_fill=True, fill_value=None):
         # but is passed thru internally
         assert isna(fill_value)
 
+        indexer = np.asarray(indexer)
+
+        # take on empty array only valid if result is an all-missing array
+        if not len(self) and not (indexer == -1).all():
+            raise IndexError("cannot do a non-empty take from an empty array.")
```

Comment:
this also needs a test

Reply:
This is already tested in the extension array tests for categorical (do you want a duplicate test in the categorical tests?)